src/FortCL.f90

```diff
@@ -37,18 +37,23 @@ module fortcl
   !===================================================
 
   !> Initialise the GOcean environment
-  subroutine ocl_env_init()
+  subroutine ocl_env_init(num_queues)
     use ocl_utils_mod, only: init_device, init_cmd_queues
     implicit none
+    integer, intent(in), optional :: num_queues
     integer :: ierr
 
-    if(cl_env_initialised)return
+    if(cl_env_initialised) return
 
     ! Initialise the OpenCL device
     call init_device(cl_device, cl_version_str, cl_context)
 
     ! Create command queue(s)
-    cl_num_queues = 4
+    if (.not.present(num_queues)) then
```

Comment:
Sorry to be picky but please could you add a check that num_queues is > 1 if it is present. Apart from that it's fine.

Reply:
Added check.